components/SearchableLocationDropdown.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  forwardRef,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { View, StyleSheet, Platform, Text } from 'react-native';
+import {
+  AutocompleteDropdown,
+  TAutocompleteDropdownItem,
+  AutocompleteDropdownRef,
+} from 'react-native-autocomplete-dropdown';
+import MapView from 'react-native-maps';
+import { Theme, useTheme } from 'react-native-paper';
+import { EdgeInsets, useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+  dropdownDebounceTime,
+  dropdownInputHeight,
+  dropdownListMaxHeight,
+} from '../constants/layout';
+import {
+  initialLatitudeDelta,
+  initialLongitudeDelta,
+  deltaRatio,
+} from '../constants/location';
+import { MapContext } from '../context';
+import { LocationItemProps } from '../models/Hotspot';
+import { Location } from '../models/Location';
+import { searchLocations } from '../utils/hotspots';
+
+export const SearchableLocationDropdown = forwardRef(
+  (_props, ref: React.ForwardedRef<MapView>) => {
+    const [loading, setLoading] = useState(false);
+    const [suggestionsList, setSuggestionsList] = useState<
+      TAutocompleteDropdownItem[] | null
+    >(null);
+    const searchRef = useRef<any>(null);
+    const dropdownController = useRef<any>(null);
+    const { selectedLocation, setSelectedLocation } = useContext(MapContext);
+
+    const insets = useSafeAreaInsets();
+    const theme = useTheme();
+    const styles = getStyles(theme, insets);
+
+    const searchForAddress = async (address: string) => {
+      try {
+        const results: any = await searchLocations(address, 'ip');
+        return results;
+      } catch (e) {
+        alert('Address not found');
+        return [];
+      }
+    };
+
+    const getSuggestions = useCallback(async (query: string) => {
+      setLoading(true);
+      const locationResults = await searchForAddress(query);
+
+      const suggestions = locationResults.map((item: LocationItemProps) => ({
+        id: `${item.coordinates?.latitude};${item.coordinates?.longitude}`,
+        title: item.placeName,
+      }));
+
+      setSuggestionsList(suggestions);
+      setLoading(false);
+    }, []);
+
+    const animateToSelectedLocation = (locationData: LocationItemProps) => {
+      if (!ref || typeof ref === 'function') return;
+
+      ref?.current?.animateToRegion({
+        latitude: locationData!.coordinates!.latitude,
+        longitude: locationData!.coordinates!.longitude,
+        latitudeDelta: initialLatitudeDelta / deltaRatio,
+        longitudeDelta: initialLongitudeDelta / deltaRatio,
+      });
+    };
+
+    const onClearPress = useCallback(() => {
+      setSuggestionsList(null);
+    }, []);
+
+    const onOpenSuggestionsList = useCallback((isOpened: any) => {}, []);
+
+    const LocationItem = ({ item }: any) => (
+      <View style={styles.locationItem}>
+        <Text style={styles.locationTitle}>{item.title}</Text>
+      </View>
+    );
```

Comment:
shouldn't `isOpened` be a boolean?
Also, you can type the props as:
```typescript
{item}: {item: {title: string}}
```

Reply:
There's a better way to write it. The `item` object has the `TAutocompleteDropdownItem` type, so I'll go with that one.